firrtl/src/main/scala/firrtl/ir/IR.scala

```diff
@@ -194,6 +194,11 @@ abstract class Literal extends Expression {
   val value: BigInt
   val width: Width
 }
+
+case class ProbeExpr(expr: Expression, tpe: Type = UnknownType) extends Expression with UseSerializer
```

Comment:
I think the `tpe` field is unnecessary (also for `ProbeRead`), it's unused. Is there a reason for it?

Reply:
it's needed because it extends `Expression`, which contains that field